### PR TITLE
feat(payments-cart): cart error handling wrapper

### PR DESCRIPTION
**Pull request by julianpoy**
## Because

- We want to easily consume our custom errors and apply them as cart error reason IDs

## This pull request

- Creates a home for which to do that transformation, and wraps many of our existing methods in this wrapper.

## Issue that this pull request solves

Closes FXA-10627

### CODE REVIEW

File: libs/payments/cart/src/lib/cart.service.ts [hunk 1]
```diff
@@ -72,6 +74,41 @@ export class CartService {
     private paymentIntentManager: PaymentIntentManager
   ) {}
 
+  /**
+   * Should be used to wrap any method that mutates an existing cart.
+   * This transforms errors into a cart error reason ID and finalizes the cart with an error.
+   */
+  private async wrapWithCartCatch<T>(cartId: string, method: () => Promise<T>) {
```
Comment (david1alvarez) [inline libs/payments/cart/src/lib/cart.service.ts:81]: Two comments here:
1. I think this should be converted to a decorator method. It inherently does not modify the original return value, but injects an "on failure" clause, which feels like it nicely matches the try/catch decorator pattern.
2. The function's name does not describe the impact of the function. Something like `failCartOnReject` might be clearer!

Comment (david1alvarez) [reply libs/payments/cart/src/lib/cart.service.ts:81]: It looks like each of the calls to `wrapWithCartCatch` wraps the entirety of the affected method's inner logic (part of why it feels very `decorator`y to me)

Comment (julianpoy) [reply libs/payments/cart/src/lib/cart.service.ts:81]: The implementation follows the described behavior in the ticket that was pointed/sliced -- given that, I'd prefer to move forward with the implementation as it is here. I'll definitely respond to these comments, but just want to make it clear I'd like to merge it as the ticket describes.

1. Moving to a decorator:
- Changing this to a decorator would inhibit my ability to use it here: https://github.com/mozilla/fxa/pull/18150/files#diff-20bf2f21059e7204e19eaa034be993bf62b2f4356de22faed1981b6bf4d37f99L256-L277
- Changing to a decorator would make it difficult to pull the `cartId` out of the method. Typings for decorators are generally rough, and given that it inherently relies on cartId from the provided method, we'd either need a context attached similar to type-cacheable, or assume the location in the originating function.
2. The name seems clear enough to me, but I'm not opposed to changing it. Given that Reino was the original creator of the ticket, lets defer to him if that's alright with you.

**Review by david1alvarez — APPROVED** [review]
marking approved, as the only comments are stylistic

**Pull request merged**